docs/developer/designs/min-runtime/README.md

```diff
@@ -0,0 +1,118 @@
+# [DRAFT] Minimum runtime before preemptions and reclaims
+
+## Overview
+
+This document proposes a new feature called "reclaim-min-runtime" and "preempt-min-runtime" that provides configurable guarantees for job runtime before preemption or reclaim can occur. This feature enables administrators to define minimum runtime guarantees at various levels of the resource hierarchy: node pool, nth level queue and leaf-queue.
+
+## Motivation
+
+Unrestrained preemption can lead to resource thrashing, where jobs are repeatedly preempted before making meaningful progress. The min-runtime feature addresses this issue by providing configurable minimum runtime guarantees that can be set by either cluster operators or sometimes within parts of the queue to provide guarantees about minimal useful work.
+
+## Detailed Design
+
+### Reclaim Min Runtime Configuration
+
+The reclaim-min-runtime parameter can be configured with the following values:
+
+- **0 (default)**: Jobs are always preemptible via reclaims
```

Comment:
n alternative way to represent this option is through a nullable configuration, which can either be set or left unset. If unset, it is interpreted as having no minimum runtime, allowing immediate reclamation.

Reply:
Unset on queue level will mean "look at parent" and eventually "look at nodepool", hence wanted to avoid overloading the meaning.